### PR TITLE
fix(shift_decider): prevent chattering of gear command when target speed is near 0

### PR DESCRIPTION
**Pull request by tkimura4**
## Description

If the target speed is around 0.01(```vel_threshold```) when Autoware engages, gear command may chatter on DRIVE and PARKING ( = ```current_gear_ptr_->report```).

Such gear command chattering can adversely affect hardware, so this PR prevents chattering by sending the same gear command as last time when the target speed is low.

image

## Tests performed
no tests

## Notes for reviewers
none

## Interface changes
none

## Effects on system behavior
This PR prevents chattering of gear command.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/shift_decider/src/shift_decider.cpp [hunk 1]
```diff
@@ -83,7 +83,7 @@ void ShiftDecider::updateCurrentShiftCmd()
     } else if (control_cmd_->longitudinal.speed < -vel_threshold) {
       shift_cmd_.command = GearCommand::REVERSE;
     } else {
-      shift_cmd_.command = current_gear_ptr_->report;
+      shift_cmd_.command = prev_shift_command;
```
Comment (shmpwk) [inline control/shift_decider/src/shift_decider.cpp:86]: Reading the PR background, the problem is occurred when the ego engages, which means shifting from parking to driving.
In that situation the change should be appried not at L86 but L95?

Comment (tkimura4) [reply control/shift_decider/src/shift_decider.cpp:86]: ```shift_cmd_.command = current_gear_ptr_->report``` at L95 is called when the vehicles does not engages.
I think L95 is OK to be as it is

Comment (shmpwk) [reply control/shift_decider/src/shift_decider.cpp:86]: Got it thank you.

**Review by shmpwk — APPROVED** [review]

**Review by TakaHoribe — APPROVED** [review]
LGTM

**Pull request merged**